### DISCUSSION TITLE
Nivo website notification "new documentation available"

### DISCUSSION THREAD

**Discussion by stahlmanDesign** [plouc/nivo#1610] [Q&A] [answered]
With SPAs it is often necessary to reload the site when a new version comes out (to avoid calling an API that may have changed).

**I have seen Nivo detects this:**

If I have [Nivo.rocks](https://nivo.rocks) opened in a tab when a new version is published, the site shows a notification

>  "new documentation available/changed"

and offers to reload the page.

@plouc How is this done? I imagine two possibilities:
- A script constantly polls the website to check if the version is different from the one loaded in memory
- An event is sent to all opened instances of the site when a new version is published
 
What are best practices?

**Comment by wyze** [comment, marked as answer]
It uses a service worker. The website is powered by gatsby. We use `gatsby-plugin-offline` which generates a service worker. Then we export the following function from the browser config:

https://github.com/plouc/nivo/blob/23059e02a430ae2cdd9814c5b99caa1d41ce7913/website/gatsby-browser.js#L20-L27

**Reply by stahlmanDesign** [reply]
Thanks. I've had problems with service workers in the past, but this looks like a good use for them.
